src/library_syscall.js

```diff
@@ -942,30 +942,36 @@ var SyscallsLibrary = {
     if (buffers[1].length) SYSCALLS.printChar(1, {{{ charCode("\n") }}});
     if (buffers[2].length) SYSCALLS.printChar(2, {{{ charCode("\n") }}});
   },
-  __syscall146__deps: ['$flush_NO_FILESYSTEM'],
+  fd_write__deps: ['$flush_NO_FILESYSTEM'],
 #if EXIT_RUNTIME == 1 && !MINIMAL_RUNTIME // MINIMAL_RUNTIME does not have __ATEXIT__ (so it does not get flushed stdout at program exit - programs in MINIMAL_RUNTIME do not have a concept of exiting)
-  __syscall146__postset: '__ATEXIT__.push(flush_NO_FILESYSTEM);',
+  fd_write__postset: '__ATEXIT__.push(flush_NO_FILESYSTEM);',
 #endif
 #endif
-  __syscall146: function(which, varargs) { // writev
+  fd_write: function(stream, iov, iovcnt, pnum) {
```

Comment:
Can we move this to library_wasi.js?  Or are we too entangled with the rest of this file to do that?

Reply:
If we library_wasi.js is too much, maybe move this to the beginning/end of this file behind a big "// WASI" comment, to logically separate the two types of syscall rather than interleave them.

Reply:
@sbc100 Hard to separate entirely, yeah. But I like @jgravelle-google's idea to move it to a separate part in the file, will put at the end.